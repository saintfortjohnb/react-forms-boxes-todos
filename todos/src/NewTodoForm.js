import React, { useState } from 'react';
import { v4 as uuid } from 'uuid';

function NewTodoForm({ addTodo }) {
  const [task, setTask] = useState('');

  const handleSubmit = (e) => {
    e.preventDefault();
    addTodo({ id: uuid(), task });
    setTask('');
  };

  return (
    <form onSubmit={handleSubmit}>
      <input 
        type="text" 
        value={task} 
        onChange={e => setTask(e.target.value)} 
      />
      <button type="submit">Add Todo</button>
    </form>
  );
}

export default NewTodoForm;
