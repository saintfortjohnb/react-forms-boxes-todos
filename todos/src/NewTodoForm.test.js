import React from 'react';
import { render } from '@testing-library/react';
import NewTodoForm from './NewTodoForm';

// Smoke Test
it('NewTodoForm renders without crashing', () => {
  const mockAddTodo = jest.fn();
  render(<NewTodoForm addTodo={mockAddTodo} />);
});

// Snapshot Test
it('NewTodoForm matches snapshot', () => {
  const mockAddTodo = jest.fn();
  const { asFragment } = render(<NewTodoForm addTodo={mockAddTodo} />);
  expect(asFragment()).toMatchSnapshot();
});
