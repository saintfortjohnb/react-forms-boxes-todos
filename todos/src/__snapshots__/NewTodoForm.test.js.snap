// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewTodoForm matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <input
      type="text"
      value=""
    />
    <button
      type="submit"
    >
      Add Todo
    </button>
  </form>
</DocumentFragment>
`;
